src/App.tsx

```diff
@@ -15,71 +15,105 @@ export const goods = [
+    return () => {
+      this.setState({ selectedGood: good });
+    };
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handlerClear = (event: { stopPropagation: () => void; }) => {
+    event.stopPropagation();
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {selectedGood === ''
+          ? (
+            <h1 className="title">No goods selected</h1>
+          ) : (
+            <h1 className="title is-flex is-align-items-center">
+              {`${this.state.selectedGood} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                onClick={this.handlerClear}
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+              />
+            </h1>
+          )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map((good: string) => {
+              return (
+                <tr
+                  data-cy="Good"
+                  className={
+                    selectedGood === good
+                      ? 'has-background-success-light'
+                      : undefined
+                  }
+                  onClick={this.handlerSelect(good)}
+                >
+                  <td>
+                    {selectedGood === good ? (
+                      <button
+                        onClick={this.handlerClear}
+                        data-cy="RemoveButton"
+                        type="button"
+                        className="button is-info"
+                      >
+                        -
+                      </button>
+                    ) : (
+                      <button
+                        onClick={this.handlerSelect(good)}
+                        data-cy="AddButton"
+                        type="button"
+                        className="button"
+                      >
+                        +
+                      </button>
+                    )}
+                  </td>
```

Comment:
for me better option for this conditional rendering it`s just change your "+" and "-", not buttons, just make handlerSelect better, add condition(check selectedGood and good what user wants to select)